draco/analysis/powerspec.py

```diff
@@ -0,0 +1,1504 @@
+    kv = u_to_kperp(vv, redshift, cosmo=cosmo)  # convert the v-array to fourier modes
+    ru = np.sqrt(ku**2 + kv**2)
+
+    # Digitize the bins
+    bin_indx = np.digitize(ru, bins=kperp_bins)
+
+    # Define empty list to store the binned 2D PS
+    ps_2D = []
+    ps_2D_w = []
+
+    # Now bin in 2D ##
+    with np.errstate(divide="ignore", invalid="ignore"):
+        for i in np.arange(len(kperp_bins) - 1) + 1:
+            p = np.nansum(w[bin_indx == i] * ps_cube[bin_indx == i]) / np.sum(
+                w[bin_indx == i]
+            )
+            ps_2D.append(p)
+            ps_2D_w.append(np.sum(w[bin_indx == i]))
+
+    return np.array(ps_2D), np.array(ps_2D_w)
+
+
+def get_1d_ps(
+    ps_2D,
+    k_perp,
+    k_para,
+    weight_cube,
+    signal_window=None,
+    Nbins_3D=10,
+    logbins_3D=True,
+):
+    """Compute the 3D spherically averaged power spectrum.
+
+    Parameters
+    ----------
+    ps_2D :  np.ndarray[kpar,kperp]
+     The cylindrically averaged 2D power spectrum.
+    k_perp : np.array[kperp]
+     The k_perp array after cylindrically binning.
+    k_para : np.array[kpar]
+     The k_parallel array.
+    weight_cube :  np.ndarray[kpar,kperp]
+      The weight array to use during spherical averaging.
+    signal_window :  np.ndarray[kpar,kperp]
+      The signal window mask.
+    Nbins_3D : int
+      The number of 3D bins
+    logbins_3D : bool
+      Bin in logarithmic space if True.
+
+    Returns
+    -------
+    k1d: np.array[Nbins_3d]
+      The K-values corresponding to the bin center
+    ps_3D: np.array[Nbins_3d]
+     The spherically avg power spectrum.
+    ps_3D_err: np.array[Nbins_3d]
+      The error in the PS (sample variance)
+    variance: np.array[Nbins_3d]
+      The variance in the power spectrum
+      estimated from thermal noise.
+    """
+    # Estimate the 1D k-modes
+    kpp, kll = np.meshgrid(k_perp, k_para)
+    k = np.sqrt(kpp**2 + kll**2)
+
+    # apply the window mask
+    if signal_window is not None:
+        k = k[signal_window]
+        ps_2D = ps_2D[signal_window]
+        w = weight_cube[signal_window]
+
+    # Take the min and max of 1D k-modes
+    kmin = k[k > 0].min()
+    kmax = k.max()
+
+    # estimat the bins
+    if logbins_3D:
+        kbins = np.logspace(np.log10(kmin), np.log10(kmax), Nbins_3D)
+    else:
+        kbins = np.linspace(kmin, kmax, Nbins_3D)
+
+    # Flatten the arrays
+    p1D = ps_2D.flatten()
+    w1D = w.flatten()
+    k1D = k.flatten()
+
+    # find the indices of bins
+    indices = np.digitize(k1D, kbins)
+
+    # Define the empty lists to save the
+    # averaged power spec and other relevant quantities
+    ps_3D = []
+    ps_3D_err = []
+    k3D = []
+    variance = []
+    # Average the modes falls in each bin
+    with np.errstate(divide="ignore", invalid="ignore"):
+        for i in np.arange(len(kbins) - 1) + 1:
+            w_b = w1D[indices == i]
+            p = np.nansum(w_b * p1D[indices == i]) / np.sum(w_b)
```

Comment:
```suggestion
            p = np.sum(w_b * p1D[indices == i]) / np.sum(w_b)
```
Same reason as above.

Reply:
Changed.